contracts/aggregator/fantom/flashloan/main.sol

```diff
@@ -0,0 +1,180 @@
+
+        emit LogFlashloan(msg.sender, _route, _tokens, _amounts);
+    }
+
+    /**
+     * @dev Function to add new routes.
+     * @notice Function to add new routes and implementations.
+     * @param _routes routes to add.
+     * @param _impls implementations of their respective routes.
+     */
+    function addNewRoutesAndEnable(uint256[] memory _routes, address[] memory _impls) public onlyOwner {
+        require(_routes.length == _impls.length, 'lengths-dont-match');
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            require(routeToImpl[_routes[i]] == address(0), 'route-already-exists');
+            routeToImpl[_routes[i]] = _impls[i];
+            routeStatus[_routes[i]] = true;
+            routes.push(_routes[i]);
+        }
+    }
+
+    /**
+     * @dev Function to update existing routes.
+     * @notice Function to update existing routes and implementations.
+     * @param _routes routes to update.
+     * @param _impls implementations of their respective routes.
+     */
+    function updateRouteImplementations(uint256[] memory _routes, address[] memory _impls) public onlyOwner {
+        require(_routes.length == _impls.length, 'lengths-dont-match');
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            routeToImpl[_routes[i]] = _impls[i];
+        }
+    }
+
+    /**
+     * @dev Function to change route status.
+     * @notice Function to enable and disable routes.
+     * @param _routes routes those status we want to change.
+     * @param _statuses new statuses.
+     */
+    function changeRouteStatus(uint256[] memory _routes, bool[] memory _statuses) public onlyOwner {
+        require(_routes.length == _statuses.length, 'lengths-dont-match');
+        uint256 length = _routes.length;
+        for (uint256 i = 0; i < length; i++) {
+            routeStatus[_routes[i]] = _statuses[i];
+        }
+    }
+
+    /**
+     * @dev Function to transfer fee to the treasury.
+     * @notice Function to transfer fee to the treasury. Will be called manually.
+     * @param _tokens token addresses for transferring fee to treasury.
+     * @param _to treasury address.
+     */
+    function transferFee(address[] memory _tokens, address _to) public onlyOwner {
+        uint256[] memory _amts = new uint256[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
+            IERC20 token_ = IERC20(_tokens[i]);
+            uint256 decimals_ = TokenInterface(_tokens[i]).decimals();
+            uint256 amtToSub_ = decimals_ == 18 ? 1e10 : decimals_ > 12 ? 10000 : decimals_ > 7 ? 100 : 10;
+            _amts[i] = token_.balanceOf(address(this)) > amtToSub_ ? (token_.balanceOf(address(this)) - amtToSub_) : 0;
+            if (_amts[i] > 0) token_.safeTransfer(_to, _amts[i]);
+        }
+        emit LogCollectRevenue(_to, _tokens, _amts);
+    }
```

Comment:
@shriyatyagii I think we can move these function to `AdminModule`

Reply:
Done